### PR TITLE
Make all render driver project settings require restart

### PR DESCRIPTION
**Pull request by CaTaTo23**
Since I tested #70315 I saw that on change renderer in project settings does not offer to restart editor, so I made them restart editor 🙂

### CODE REVIEW

**Review by Calinou — APPROVED** [review]

**Comment by AThousandShips** [comment]
Can you please update the comment message to match the PR title, more correct terminology, use `got commit --amend` and then `git push --force`

**Comment by KoTeYkA23** [comment]
done

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
I just merged #72831 which introduced some merge conflicts as it's changing the same project settings. Could you rebase to solve the merge conflicts?

**Comment by KoTeYkA23** [comment]
done

**Comment by AThousandShips** [comment]
Please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

**Comment by KoTeYkA23** [comment]
yeah, sorry insead rebase I merged it accidently, hopefully I fixes everything

**Comment by KoTeYkA23** [comment]
oh, I accidentally remove angle hints, one sec

File: main/main.cpp [hunk 1]
```diff
@@ -1720,14 +1720,13 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
...
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.macos", PROPERTY_HINT_ENUM, driver_hints_angle), default_driver);
-
+		GLOBAL_DEF_RST("rendering/gl_compatibility/driver", default_driver);
+		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.windows", PROPERTY_HINT_ENUM, driver_hints), driver_hints_angle);
+		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.linuxbsd", PROPERTY_HINT_ENUM, driver_hints), default_driver);
+		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.web", PROPERTY_HINT_ENUM, driver_hints), default_driver);
+		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.android", PROPERTY_HINT_ENUM, driver_hints), default_driver);
+		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.ios", PROPERTY_HINT_ENUM, driver_hints), default_driver);
+		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.macos", PROPERTY_HINT_ENUM, driver_hints), driver_hints_angle);
```
Comment (AThousandShips) [inline main/main.cpp:1729]:
```suggestion
		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.macos", PROPERTY_HINT_ENUM, driver_hints_angle), default_driver);
```

File: main/main.cpp [hunk 2]
```diff
@@ -1720,14 +1720,13 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
...
-		GLOBAL_DEF("rendering/gl_compatibility/driver", default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.windows", PROPERTY_HINT_ENUM, driver_hints_angle), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.linuxbsd", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.web", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.android", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.ios", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.macos", PROPERTY_HINT_ENUM, driver_hints_angle), default_driver);
-
+		GLOBAL_DEF_RST("rendering/gl_compatibility/driver", default_driver);
+		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.windows", PROPERTY_HINT_ENUM, driver_hints), driver_hints_angle);
```
Comment (AThousandShips) [inline main/main.cpp:1724]:
```suggestion
		GLOBAL_DEF_RST(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.windows", PROPERTY_HINT_ENUM, driver_hints_angle), default_driver);
```

**Comment by AThousandShips** [comment]
You changed the wrong part of the code

**Comment by AThousandShips** [comment]
Please squash again :)

**Comment by KoTeYkA23** [comment]
done, sorry for newbie git mistakes

**Comment by AThousandShips** [comment]
It's a learning process :)

**Comment by akien-mga** [comment]
Technically, most of these settings don't really require a restart. A restart is only needed for the setting to be picked up for the editor process, but it will pick up the changed setting fine when running the game. Only the platform override which matches the editor's host platform also really warrants restarting to pick it up in the editor.

Still, it's not a bad idea to suggest a restart when changing those. I'm just slightly worried about the UX - when you change those settings, you typically need to change several of them (e.g. all platform specific overrides), and having a nagging message that one should restart the editor after you change the first, when you're meant to change several of them, might be a bit annoying. But maybe it's fine.

**Comment by KoTeYkA23** [comment]
> Technically, most of these settings don't really require a restart. A restart is only needed for the setting to be picked up for the editor process, but it will pick up the changed setting fine when running the game. Only the platform override which matches the editor's host platform also really warrants restarting to pick it up in the editor.

Still, it's not a bad idea to suggest a restart when changing those. I'm just slightly worried about the UX - when you change those settings, you typically need to change several of them (e.g. all platform specific overrides), and having a nagging message that one should restart the editor after you change the first, when you're meant to change several of them, might be a bit annoying. But maybe it's fine.

Well editor not closing forcefully, only suggest to restart, and if Godot have platform specific preprocessors we could just define restartable options only on current platform, we could ofk just platform check with default if, but I think this will be unnecessary  check

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**